### PR TITLE
Upgrade terraform-provider-aws to v5.32.1

### LINKED ISSUE #3253: Upgrade terraform-provider-aws to v5.32.0
Release details: https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.32.0

### LINKED ISSUE #3254: Upgrade terraform-provider-aws to v5.32.1
Release details: https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.32.1

### PR DESCRIPTION
**Pull request by iwahbe**
This PR was generated via `$ upgrade-provider pulumi/pulumi-aws`.

---

- Upgrading terraform-provider-aws from 5.31.0  to 5.32.1.
	Fixes #3254
	Fixes #3253

### CODE REVIEW

**Review by blampe — APPROVED** [review]

**Comment by blampe** [comment]
@iwahbe 

> panic: no AWS SDK v1 API client factory: s3

hmm

**Comment by iwahbe** [comment]
> @iwahbe
> 
> > panic: no AWS SDK v1 API client factory: s3
> 
> hmm

I see it. That is used in an old part of our legacy s3 bucket implementation, and I needed to extend the patch to keep it compiling. It sounds like I need to take another look there.

**Comment by iwahbe** [comment]
> Intending to finish this up or handing it over? Thanks. That's the only Tier 1 provider upgrade requested at the moment.

We just discussed and I will be handing over. Documenting here for continuity.

File: provider/doc_edits.go [hunk 1]
```diff
@@ -186,6 +186,13 @@ func (r replacementFile) checkForTODOs(path string, content []byte) {
 		start, end = findLine(content, m[0])
 		line := string(content[start:end])
 
+		// Do not add duplicate replacements to requirements.json
```
Comment (t0yv0) [inline provider/doc_edits.go:189]: Oof. I didn't dissect his PR into commits and missed this so ended up reimplementing.

**Pull request closed without merging**